nixos/modules/services/web-apps/mattermost.nix

```diff
@@ -261,73 +428,148 @@ in
 
   config = mkMerge [
     (mkIf cfg.enable {
-      users.users = optionalAttrs (cfg.user == "mattermost") {
-        mattermost = {
-          group = cfg.group;
-          uid = config.ids.uids.mattermost;
-          home = cfg.statePath;
-        };
+      users.users.${cfg.user} = {
+        group = cfg.group;
+        uid = config.ids.uids.mattermost;
+        home = cfg.dataDir;
```

Comment:
Thanks, I think there was a reason this was added from the blame and I'll change it back.

Reply:
I think I figured it out. The secret is to use `uid = mkIf (cfg.user == "mattermost") config.ids.uids.mattermost;` since then we auto-assign a UID if and only if the username is nonstandard. Same with gid. Without this, my tests with a nonstandard username failed because there was no user created.

Reply:
I don't know how many NixOS Mattermost deployments there are or how many of them are changing the user, but suppose someone was running it as their own personal user. With the previous behavior, no user configuration is done. With the current behavior, this would cause a conflict on `group` and `home` requiring `mkForce` or similar.

Maybe there needs to be a release notes item to the effect of "if you run mattermost with a nonstandard user, you should watch out/review the changes"? I still have no idea whether we're breaking anyone with this or just fixing a potential bug.